### PR TITLE
Create competition model

### PR DESCRIPTION
**Pull request by smarzban**
Competition Model:
- Name
- Opening and closing date times
- Competition Status: 
       \* Upcoming
       \* Currently running
       \* Closed

### CODE REVIEW

**Comment by smarzban** [comment]
@miltontony ready for review

**Comment by codiebeulaine** [comment]
@miltontony ready for review

File: molo/yourwords/models.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+from django.db import models
+
+
+class competition(models.Model):
```
Comment (miltontony) [inline molo/yourwords/models.py]: naming convention - class should be CamelCase

**Comment by codiebeulaine** [comment]
@miltontony ready for review

File: molo/yourwords/models.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+from wagtail.wagtailcore.models import Page
+
+
+class competition(Page):
```
Comment (miltontony) [inline molo/yourwords/models.py]: Naming convention - it should be `Competition(Page)`

File: molo/yourwords/models.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    name = Page.CharField(max_length=128)
+    startDate = Page.DateTimeField(null=True, blank=True)
```
Comment (miltontony) [inline molo/yourwords/models.py]: Naming convention - `field_name_with_underscores` it should be `start_date`

File: molo/yourwords/models.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    endDate = Page.DateTimeField(null=True, blank=True)
```
Comment (miltontony) [inline molo/yourwords/models.py]: You're also missing a few fields we discussed from the SoW - description and body

File: molo/yourwords/models.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+
+class competition_entry(Page):
+    story_name = Page.CharField(max_length=128)
+    story_text = Page.TextField()
+    TERMS_OR_CONDITIONS_CHOICES = (
```
Comment (miltontony) [inline molo/yourwords/models.py]: Yes and No should be boolean fields (same for `hide_real_name` below)

**Comment by codiebeulaine** [comment]
@miltontony ready for review

File: molo/yourwords/models.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
+from wagtail.wagtailcore.models import Page
+
+
+class Competition(Page):
+    name = Page.CharField(max_length=128)
```
Comment (miltontony) [inline molo/yourwords/models.py]: There's already a `title` field in wagtail so you don't need this

File: molo/yourwords/models.py [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    description = Page.TextField()
+    content = Page.TextField()
```
Comment (miltontony) [inline molo/yourwords/models.py]: Please use the [StreamField](https://github.com/praekelt/molo/blob/develop/molo/core/models.py#L267-L274) what we use in the ArticlePage for this field

File: molo/yourwords/models.py (hunk 6 above)
Comment (codiebeulaine) [inline molo/yourwords/models.py]: Will take it out, does Django not need this field though? Or will it come up automatically?

File: molo/yourwords/models.py [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    start_date = Page.DateTimeField(null=True, blank=True)
+    end_date = Page.DateTimeField(null=True, blank=True)
+
+
+class CompetitionEntry(Page):
+    story_name = Page.CharField(max_length=128)
+    story_text = Page.TextField()
+    TERMS_OR_CONDITIONS_CHOICES = (
```
Comment (miltontony) [inline molo/yourwords/models.py]: You don't need these choices anymore - a boolean field is either True/False. (same for the hide real name below)

**Comment by codiebeulaine** [comment]
cool, :+1: ?

File: molo/yourwords/models.py [hunk 9]
```diff
@@ -0,0 +1,15 @@
+from wagtail.wagtailcore.models import Page
+
+
+class Competition(Page):
```
Comment (miltontony) [inline molo/yourwords/models.py]: I think you should also add the `terms_and_conditions` page like we discussed using [this example](https://github.com/praekelt/molo/blob/develop/molo/core/models.py#L50-L57).

**Comment by codiebeulaine** [comment]
@miltontony like this?

**Comment by miltontony** [comment]
Yes, so far so good. You now need to run `./manage.py makemigrations` to add migrations for the models

**Comment by miltontony** [comment]
:+1:  looks good. On to the next one

**Comment by smarzban** [comment]
@miltontony ready for review :)
competition
competition_page

**Comment by miltontony** [comment]
:+1:

**Pull request merged**